ethereum/contracts/pyth/PythGovernanceInstructions.sol

```diff
@@ -0,0 +1,157 @@
+        require(magic == MAGIC, "invalid magic for GovernanceInstruction");
+        index += 4;
+
+        uint8 modNumber = encodedInstruction.toUint8(index);
+        gi.module = GovernanceModule(modNumber);
+        index += 1;
+
+        require(gi.module == MODULE, "invalid module for GovernanceInstruction");
+
+        uint8 actionNumber = encodedInstruction.toUint8(index);
+        gi.action = GovernanceAction(actionNumber);
+        index += 1;
+
+        gi.targetChainId = encodedInstruction.toUint16(index);
+        index += 2;
+
+        gi.payload = encodedInstruction.slice(index, encodedInstruction.length - index);
+    }
+
+    /// @dev Parse a UpgradeContractPayload (action 1) with minimal validation
+    function parseUpgradeContractPayload(bytes memory encodedPayload) public pure returns (UpgradeContractPayload memory uc) {
+        uint index = 0;
+
+        uc.newImplementation = address(encodedPayload.toAddress(index));
+        index += 20;
+
+        require(encodedPayload.length == index, "invalid length for UpgradeContractPayload");
+    }
+
+    /// @dev Parse a SetGovernanceDataSourcePayload (action 2) with minimal validation
+    function parseSetGovernanceDataSourcePayload(bytes memory encodedPayload) public pure returns (SetGovernanceDataSourcePayload memory sgds) {
+        uint index = 0;
+
+        sgds.newGovernanceDataSource.chainId = encodedPayload.toUint16(index);
+        index += 2;
+
+        sgds.newGovernanceDataSource.emitterAddress = encodedPayload.toBytes32(index);
+        index += 32;
+
+        sgds.initialSequence = encodedPayload.toUint64(index);
+        index += 8;
+
+        require(encodedPayload.length == index, "invalid length for SetGovernanceDataSourcePayload");
+    }
+
+    /// @dev Parse a SetDataSourcesPayload (action 3) with minimal validation
+    function parseSetDataSourcesPayload(bytes memory encodedPayload) public pure returns (SetDataSourcesPayload memory sds) {
+        uint index = 0;
+
+        uint8 dataSourcesLength = encodedPayload.toUint8(index);
+        index += 1;
+
+        sds.dataSources = new PythInternalStructs.DataSource[](dataSourcesLength);
+
+        for(uint i = 0; i < dataSourcesLength; i++) {
+            sds.dataSources[i].chainId = encodedPayload.toUint16(index);
+            index += 2;
+
+            sds.dataSources[i].emitterAddress = encodedPayload.toBytes32(index);
+            index += 32;
+        }
+
+        require(encodedPayload.length == index, "invalid length for SetDataSourcesPayload");
+    }
+
+    /// @dev Parse a SetFeePayload (action 4) with minimal validation
+    function parseSetFeePayload(bytes memory encodedPayload) public pure returns (SetFeePayload memory sf) {
+        uint index = 0;
+
+        uint64 val = encodedPayload.toUint64(index);
+        index += 8;
+
+        uint64 expo = encodedPayload.toUint64(index);
+        index += 8;
+
+        sf.newFee = uint256(val) * uint256(10)**uint256(expo);
```

Comment:
I think we need an overflow check here.

Reply:
From solidity 0.8 onwards all math operations in solidity are checked. (unless you do it unchecked)